Library/ViewModels/ChangePasswordViewModel.swift

```diff
@@ -117,30 +109,35 @@ ChangePasswordViewModelInputs, ChangePasswordViewModelOutputs {
     self.onePasswordFindPasswordForURLString = self.onePasswordButtonTappedProperty.signal
       .map { AppEnvironment.current.apiService.serverConfig.webBaseUrl.absoluteString }
 
-    let requirementsMet = Signal.combineLatest(passwordsMatch, lengthMeetsReq)
-      .map { $0.0 && $0.1 }
+    let newPasswordValidationText = fieldLengthIsValid
+      .map(passwordValidationText)
+      .skipRepeats()
 
-    self.validationErrorLabelIsHidden = requirementsMet.skipRepeats()
+    let newPasswordAndConfirmationValidationText = Signal.combineLatest(fieldLengthIsValid, fieldsMatch)
+      .map(passwordValidationText)
+      .skipRepeats()
 
-    let errorLabelMessage = Signal.combineLatest(passwordsMatch, lengthMeetsReq)
-      .map { requirements -> String? in
-        if !requirements.1 {
-          return Strings.Password_min_length_message()
-        } else if !requirements.0 {
-          return Strings.Passwords_matching_message()
-        } else {
-          return nil
-        }
-    }
+    let validationLabelText = Signal.merge(
+      self.viewDidAppearProperty.signal.mapConst(String()),
+      newPasswordValidationText,
+      newPasswordAndConfirmationValidationText
+    )
 
-    self.validationErrorLabelMessage = errorLabelMessage
-      .skipNil()
-      .skipRepeats()
+    self.validationErrorLabelMessage = validationLabelText
+      .map { $0 ?? String() }
+
+    let validationLabelTextIsEmpty = self.validationErrorLabelMessage
+      .map { $0.isEmpty }
+    .logEvents(identifier: "*** validationLabelTextIsEmpty")
```

Comment:
We should remove logging here

Reply:
![giphy](https://media.giphy.com/media/jUwpNzg9IcyrK/giphy.gif)